src/main/java/net/dries007/tfc/common/recipes/TypedRecipeSerializer.java

```diff
@@ -0,0 +1,15 @@
+/*
+ * Licensed under the EUPL, Version 1.2.
+ * You may obtain a copy of the Licence at:
+ * https://joinup.ec.europa.eu/collection/eupl/eupl-text-eupl-12
+ */
+
+package net.dries007.tfc.common.recipes;
+
+import net.minecraft.item.crafting.IRecipe;
+import net.minecraft.item.crafting.IRecipeType;
+
+public abstract class TypedRecipeSerializer<R extends IRecipe<?>> extends RecipeSerializer<R>
```

Comment:
Is there a reason this class exists at all, and not just coded to `KnappingRecipe.Serializer` ?

Reply:
Yes RockKnappingRecipe extends this instead of extending the default serializer. I had a lot of trouble with generic types and this was the compromise